pkg/infra/provider.go

```diff
@@ -388,3 +394,34 @@ func (p *AWSProvider) getClusterSecret(ctx context.Context) (*corev1.Secret, err
 
 	return secret, nil
 }
+
+func (p *AWSProvider) reconcileKubeconfig(ctx context.Context) error {
+	scopeCluster := p.scope
+
+	cluster := scopeCluster.Cluster
+	if cluster.Status.Phase != string(clusterv1alpha1.ClusterPhaseReady) {
```

Comment:
if this only reconcile when cluster become ready, it's better to move to cluster_controller.go L230?

Reply:
I donot think it make sense. Different implementor can have different kubeconfig set.

Reply:
we can add func `Kubeconfig()` on Provider?

Reply:
We have a generic `Reconcile(ctx context.Context) error` now, i donot want to add any concrete one unless we need a refactor to replace a generic one with multi concrete ones, reconcileXXX, reconcileYYY. At least for now,  I am not convinced to do that.

Reply:
I do same things for `ServiceAccountIssuer` in https://github.com/kurator-dev/kurator/pull/185, I'm trying to avoid modify `clusterv1alpha1.Cluster` out of `cluster_controller.go`.